### PR TITLE
Replace RELEASE.md with an issue template

### PR DESCRIPTION
**Pull request by marquiz**
In addition to just moving the content, update release process. Adjust
to the latest prow-based workflow. Capture steps required for
documentation.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/new-release.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+name: New Release
+about: Propose a new release
+title: Release v0.x.0
+assignees: marquiz
```
Comment (ArangoGutierrez) [inline .github/ISSUE_TEMPLATE/new-release.md:5]: Can we have multiple assignees? like Prow does?

Comment (marquiz) [reply .github/ISSUE_TEMPLATE/new-release.md:5]: Yes, I suppose

Comment (ArangoGutierrez) [reply .github/ISSUE_TEMPLATE/new-release.md:5]: ping ^

Comment (marquiz) [reply .github/ISSUE_TEMPLATE/new-release.md]: Added all (active) OWNERS, i.e. approvers and reviewers

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**